src/search_index.rs

```diff
@@ -215,6 +215,17 @@ impl SearchIndex {
                         Ok(content) => content,
                         Err(_) => return, // Skip files we can't read as text
                     };
+
+                    // Skip files with lines longer than 8096 bytes
+                    const MAX_LINE_LENGTH: usize = 8096;
+                    if content.lines().any(|line| line.len() > MAX_LINE_LENGTH) {
```

Comment:
I think we should add a filesize limit as well (unless there is already one), let's say up to 512kb